azkaban-common/src/test/java/azkaban/executor/container/KubernetesContainerizedImplTest.java

```diff
@@ -617,6 +632,92 @@ public void testVersionSetConstructionWithRampupManager() throws Exception {
+    Props currentNodeProps1 = new Props();
+    Props currentNodeJobProps1 = new Props();
+    currentNodeProps1.put("user.to.proxy", "testUser1");
+    currentNodeJobProps1.put("user.to.proxy", "");
+    when(projectManager.getProject(flow.getProjectId())).thenReturn(project);
+    when(project.getFlow(flow.getFlowId())).thenReturn(flowObj);
+    when(projectManager.getProperties(project, flowObj, node1.getId(), node1.getJobSource()))
+        .thenReturn(currentNodeProps1);
+    //when(currentNodeProps1.getString("user.to.proxy", "")).thenReturn("testUser1");
+    when(projectManager.getJobOverrideProperty(project, flowObj, node1.getId(),
+        node1.getJobSource()))
+        .thenReturn(currentNodeJobProps1);
+    //when(currentNodeJobProps1.getString("user.to.proxy", "")).thenReturn("");
+    populateProxyUsersForFlow(node1, flowObj, project, projectManager, proxyUsers);
+
+    // First test when there's no job override user.
+    Assert.assertTrue(proxyUsers.contains("testUser1"));
+    Assert.assertEquals(1, proxyUsers.size());
+    proxyUsers.clear();
+
+    // Test adding a empty string user
+    currentNodeProps1.put("user.to.proxy", "");
+    populateProxyUsersForFlow(node1, flowObj, project, projectManager, proxyUsers);
+    Assert.assertEquals(0, proxyUsers.size());
+    //when(currentNodeJobProps1.getString("user.to.proxy", "")).thenReturn("overrideUser");
+    currentNodeJobProps1.put("user.to.proxy", "overrideUser");
+    populateProxyUsersForFlow(node1, flowObj, project, projectManager, proxyUsers);
+
+    // Second test when there is a job override user.
+    Assert.assertTrue(proxyUsers.contains("overrideUser"));
+    Assert.assertEquals(1, proxyUsers.size());
+
+    // Third test : Adding a second node and testing size of proxy user list to test it has
+    // overrideUser and testUser2
+    ExecutableNode node2 = new ExecutableNode();
+    node2.setId("node2");
+    node2.setJobSource("job2");
+    node2.setStatus(Status.PREPARING);
+    Props currentNodeProps2 = new Props();
+    Props currentNodeJobProps2 = new Props();
+    when(projectManager.getProperties(project, flowObj, node2.getId(), node2.getJobSource()))
+        .thenReturn(currentNodeProps2);
+    currentNodeProps2.put("user.to.proxy", "testUser2");
+    currentNodeJobProps2.put("user.to.proxy", "");
+    //when(currentNodeProps2.getString("user.to.proxy", "")).thenReturn("testUser2");
+
+    when(projectManager.getJobOverrideProperty(project, flowObj, node2.getId(),
+        node2.getJobSource()))
+        .thenReturn(currentNodeJobProps2);
+    populateProxyUsersForFlow(node2, flowObj, project, projectManager, proxyUsers);
+
+    Assert.assertTrue(proxyUsers.contains("testUser2"));
+    Assert.assertEquals(2, proxyUsers.size());
+  }
+
+  @Test
+  public void testPopulatingJobTypeUsersForFlow() throws Exception {
+    Set<String> proxyUsers;
+    TreeSet<String> jobTypes = new TreeSet<>();
+    jobTypes.add("jobtype1");
```

Comment:
Can be done in a subsequent PR but perhaps the most useful thing to test here is `parseJobTypeUsersForFlow()`, making sure that spaces, empty string and malformed strings won't break the code.